lib/lightning/projects.ex

```diff
@@ -286,22 +287,40 @@ defmodule Lightning.Projects do
       {:error, %Ecto.Changeset{}}
 
   """
-  def update_project(%Project{} = project, attrs) do
+  def update_project(%Project{} = project, attrs, user \\ nil) do
     changeset = Project.changeset(project, attrs)
 
-    case Repo.update(changeset) do
-      {:ok, updated_project} ->
+    Multi.new()
+    |> Multi.update(:project, changeset)
+    |> maybe_audit_changes(changeset, user)
+    |> Repo.transaction()
+    |> case do
+      {:ok, %{project: updated_project}} ->
         if retention_setting_updated?(changeset) do
           send_data_retention_change_email(updated_project)
         end
 
         {:ok, updated_project}
 
-      error ->
-        error
+      {:error, :project, changeset, _changes_so_far} ->
+        {:error, changeset}
+
+      # 2024-10-29 Not tested at module-level
+      # due to the difficulty of simulating a failure
+      # without mocking
+      {:error, _operation, changeset, _changes_so_far} ->
+        {:audit_creation_error, changeset}
```

Comment:
a more idiomatic way would be to follow the pattern `{:error, reason}` where the reason could be an atom if doesn't look at the error of the changeset.